### PR TITLE
Add instructions in CONTRIBUTING.md on how to contribute new metrics

### PR DESCRIPTION
**Pull request by singku**
**Description:** 
Add instructions in CONTRIBUTING.md on how to contribute new metrics

**Link to tracking Issue:** 
https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/11025

**Testing:** 
Doc updte.

**Documentation:** 
CONTRIBUTING.md

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -78,6 +78,21 @@ The following GitHub users are the currently available sponsors, either by being
...
+Following these steps for contributing additional metrics.
+ - Make sure you have basic tools, like `golang, make and git` installed in your system. 
+ - Fork this repo and check it out locally to your system, e.g. into the directory `opentelemetry-collector-contrib`
+ - Make sure $GOPATH is added to $PATH, if not sure, run `export PATH=$PATH:$(go env GOPATH)/bin`
+ - Run `cd opentelemetry-collector-contrib`
+ - Run `make install-tools`
+ - Edit `metadata.yaml` of your metrics receiver to add new metrics, e.g.: `redisreceiver/metadata.yaml`
+ - To generate new metrics on top of this updated YAML file.
+   - Run `cd receiver/redisreceiver`
+   - Run `go generate`
```
Comment (codeboten) [inline CONTRIBUTING.md:92]:
```suggestion
   - Run `go generate ./...`
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -78,6 +78,21 @@ The following GitHub users are the currently available sponsors, either by being
 * [@djaglowski](https://github.com/djaglowski)
 
 Whenever a sponsor is picked from the top of this list, please move them to the bottom.
+
+## Adding New Metrics
```
Comment (codeboten) [inline CONTRIBUTING.md:82]:
```suggestion
## Adding metrics to existing receivers
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -78,6 +78,21 @@ The following GitHub users are the currently available sponsors, either by being
...
+Following these steps for contributing additional metrics.
```
Comment (codeboten) [inline CONTRIBUTING.md:83]: It might be worth adding a note that this workflow only applies to receivers that use mdatagen

**Review by codeboten — COMMENTED** [review]
Thanks for the contribution! I've added a couple of suggestions. @dmitryax @djaglowski please review and see if more information should be added.

File: CONTRIBUTING.md (hunk 1 above)
Comment (singku) [reply CONTRIBUTING.md:92]: I meant to only re-generate this changed receiver

File: CONTRIBUTING.md (hunk 3 above)
Comment (singku) [reply CONTRIBUTING.md]: added a note

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -78,6 +78,21 @@ The following GitHub users are the currently available sponsors, either by being
...
 
 Whenever a sponsor is picked from the top of this list, please move them to the bottom.
+
+## Adding metrics to existing receivers
+Following these steps for contributing additional metrics to existing receivers.
+ - Make sure you have basic tools, like `golang, make and git` installed in your system. 
+ - Fork this repo and check it out locally to your system, e.g. into the directory `opentelemetry-collector-contrib`
+ - Make sure $GOPATH is added to $PATH, if not sure, run `export PATH=$PATH:$(go env GOPATH)/bin`
+ - Run `cd opentelemetry-collector-contrib`
+ - Run `make install-tools`
```
Comment (mx-psi) [inline CONTRIBUTING.md:88]: Should we link to these instructions instead? https://github.com/open-telemetry/opentelemetry-collector/blob/main/CONTRIBUTING.md#fork I believe they cover the same thing; if something is missing we should add it to that document

File: CONTRIBUTING.md (hunk 1 above)
Comment (codeboten) [reply CONTRIBUTING.md:92]: Right, for some receivers (ie `hostmetricsreceiver`) the metadata file exists in subdirectories.

File: CONTRIBUTING.md (hunk 4 above)
Comment (singku) [reply CONTRIBUTING.md:88]: Did notice there was the similar part in that repository. I will update this doc with links then.

**Review by mx-psi — APPROVED** [review]

**Review by djaglowski — APPROVED** [review]

**Pull request merged**